contracts/OffsetHelper.sol

```diff
@@ -235,26 +278,31 @@ contract OffsetHelper is OffsetHelperStorage {
 
     /**
      * @notice Retire carbon credits using the lowest quality (oldest) TCO2
-     * tokens available from the specified Toucan token pool by sending MATIC.
-     * All provided MATIC is consumed for offsetting.
+     * tokens available from the specified Toucan token pool by sending native tokens e.g., MATIC.
+     * All provided native tokens  is consumed for offsetting.
      *
      * This function:
      * 1. Swaps the Matic sent to the contract for the specified pool token.
      * 2. Redeems the pool token for the poorest quality TCO2 tokens available.
      * 3. Retires the TCO2 tokens.
      *
-     * @param _poolToken The address of the Toucan pool token that the
-     * user wants to use, for example, NCT or BCT.
+     * @dev This function is only available on Polygon, not on Celo.
+     *
+     * @param _poolToken The address of the pool token to swap for,
+     *  e.g., NCT
      *
      * @return tco2s An array of the TCO2 addresses that were redeemed
      * @return amounts An array of the amounts of each TCO2 that were redeemed
      */
-    function autoOffsetExactInETH(address _poolToken)
+    function autoOffsetExactInETH(
```

Comment:
Instead of exposing a different interface based on the chain, we could have a single interface and discover in the background how a user wants to do the swap in non-Celo networks by inspecting `msg.value`:
* If `msg.value` is defined, then we know the user is trying to swap from MATIC
* If `msg.value` is not defined, then we know the user is trying to swap from an ERC20 token
* If both `msg.value` and an ERC20 address are provided as part of the swap, we should revert

Reply:
Not sure I understand:
So, instead of having a function that is called `autoOffsetExactInETH` and one that is called `autoOffsetExactInToken`  we would only have one function called `autoOffsetExactIn(_fromToken: address, _poolToken: address, _amount: uint`. 

But then how do we do the swapToken optional. It's not the best practice to have to input an empty string.
And Uniswap functions etc. differentiate to the outside if the native token is used.